ext/json/json.c

```diff
@@ -31,6 +31,14 @@
 #include "php_json.h"
 #include <zend_exceptions.h>
 
+#include <float.h>
+#if defined(DBL_MANT_DIG) && defined(DBL_MIN_EXP)
+#define NUM_BUF_SIZE (3 + DBL_MANT_DIG - DBL_MIN_EXP)
+#else
+#define NUM_BUF_SIZE 1080
```

Comment:
@TazeTSchnitzel according with http://tigcc.ticalc.org/doc/float.html the value if the constants are defined is `3 + 16 - -999` = 1018. 
There is research from @bukka and he explains on [this comment](https://github.com/php/php-src/pull/642#issuecomment-50283072):
> You might notice that the default value is 1080. Not sure where I got 1089 - I just tested the value and it's 1077 on my platform and max value that I googled was 1079 so it should never be higher than 1080. I'm almost sure that if it does, the constants will be defined float.h so it won't be a problem...